app/controllers/CheckYourAnswersController.scala

```diff
@@ -52,10 +58,28 @@ class CheckYourAnswersController @Inject() (
       renderer.render("checkYourAnswers.njk", json).map(Ok(_))
   }
 
-  def onSubmit: Action[AnyContent] = (identify andThen getData andThen requireData) {
+  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      // TODO - send answers to backend
-      Redirect(routes.BalanceConfirmationController.onPageLoad())
+      request.userAnswers.get(GuaranteeReferenceNumberPage) match {
+        case None => Future.successful(Redirect(routes.SessionExpiredController.onPageLoad()))
+        case Some(guaranteedReferenceNumber: String) =>
+          checkRateLimit(request.eoriNumber, guaranteedReferenceNumber).flatMap {
+            lockFree =>
+              if (lockFree) {
+                Future.successful(
+                  Redirect(routes.BalanceConfirmationController.onPageLoad())
+                ) //todo this needs to be completed once the back end direction has been confirmed
+              } else {
+                Future.successful(Redirect(routes.RateLimitController.onPageLoad()))
+              }
+          }
+      }
+  }
+
+  private def checkRateLimit(eoriNumber: String, guaranteedReferenceNumber: String): Future[Boolean] = {
+    val lockId   = (eoriNumber + guaranteedReferenceNumber.trim.toLowerCase).hashCode.toString
+    val duration = config.rateLimitDuration.seconds
+    mongoLockRepository.takeLock(lockId, eoriNumber, duration)
```

Comment:
@kenneth-rayner I think I might have given bad advice here when I was talking to @andydow311 about this - I've realised since we spoke that using "eoriNumber" as the owner will mean that the same EORI will be able to take the lock even if it's currently locked.
I think it might be better to use something unique like the request ID as the "owner" rather than the EORI.

Reply:
I might be wrong about this as it's not very well documented on the hmrc-mongo repo but it's worth testing for this particular possibility.